store/inmem/InMem.go

```diff
@@ -80,6 +100,48 @@ func (i *InMem) Delete(key model.Key) (store.OwnableItem, error) {
 	if !ok {
 		return store.OwnableItem{}, store.SanitizeError(store.ItemOperationError{Err: store.ErrItemNotFound, Key: key, Operation: "delete"})
 	}
+
+	if i.itemExpired(item) {
+		i.pruneExpiredItem(key.Bucket, key.ID, bucket)
+		return store.OwnableItem{}, store.SanitizeError(store.ItemOperationError{Err: store.ErrItemNotFound, Key: key, Operation: "delete"})
+	}
 	delete(bucket, key.ID)
-	return item, nil
+	if len(bucket) == 0 {
+		delete(i.data, key.Bucket)
+	}
+	return item.OwnableItem, nil
+}
+
+func (i *InMem) pruneExpiredItem(bucketName string, itemID string, bucket map[string]expireableItem) {
```

Comment:
Since we don't check that this is expired, I suggest renaming to `pruneItem` (or moving the expired check here).